hail/src/main/scala/is/hail/expr/ir/IR.scala

```diff
@@ -94,8 +94,14 @@ object If {
 
 final case class If(cond: IR, cnsq: IR, altr: IR) extends IR
 
-final case class AggLet(name: String, value: IR, body: IR) extends IR
-final case class Let(name: String, value: IR, body: IR) extends IR
+trait LetNode extends IR {
```

Comment:
I have to say, I don't super like this.  We smear env and aggEnv in various places, and mostly that seems like a recipe for error, and this encourages it.

Reply:
good point.